src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -1877,6 +1960,29 @@ class MutateSomePartColumnsTask : public IExecutableTask
 
         ctx->new_data_part->checksums = ctx->source_part->checksums;
 
+        /// Remove dropped secondary indices from checksums, to avoid complain of broken parts.
+        if ((*ctx->data->getSettings())[MergeTreeSetting::secondary_indices_on_columns_alter_modify] ==
```

Comment:
Is it possible to use variable `settings` from l. 1870?

Reply:
thanks, yes it would be more consistent inside this function.